lib/kubeclient/google_application_default_credentials.rb

```diff
@@ -5,7 +5,13 @@ module Kubeclient
   class GoogleApplicationDefaultCredentials
     class << self
       def token
-        require 'googleauth'
+        begin
+          require 'googleauth'
+        rescue LoadError
+          puts "Gem 'googleauth' not found. Kubeclient does not include the googleauth gem, "\
+          'you must include it in your own application'
```

Comment:
:+1: on the idea.  Not certain `puts` is ideal but don't know what's better.
I suggest adding a clarification to the that specifically "to support auth-provider `gcp`, you must include it ..."

Reply:
I suppose it's a specific enough thing that we can make our own error class for this